caikit/runtime/service_generation/create_service.py

```diff
@@ -115,15 +115,15 @@ def _group_modules_by_task(
 ) -> Dict[Type[TaskBase], List[CaikitMethodSignature]]:
     task_groups = {}
     for ck_module in modules:
-        if ck_module.TASK_CLASS:
-            ck_module_task_name = ck_module.TASK_CLASS.__name__
+        for task_class in ck_module.TASK_CLASSES:
+            ck_module_task_name = task_class.__name__
             if ck_module_task_name is not None:
                 for (
                     input_streaming,
                     output_streaming,
                     signature,
-                ) in ck_module._INFERENCE_SIGNATURES:
-                    task_groups.setdefault(ck_module.TASK_CLASS, {}).setdefault(
+                ) in ck_module._TASK_INFERENCE_SIGNATURES[task_class]:
```

Comment:
oof, yeah this was always a kludge here to begin with. I'd much rather be accessing public properties of the module class when we're outside of the `caikit.core.modules` package.

Reply:
I believe in python you can set class-level `@property`s by sticking them on the meta-class. So something like this in `caikit/core/modules/meta.py` should work:

```
class _ModuleBaseMeta(abc.ABCMeta):
    ...

    @property
    def tasks(cls) -> Set[TaskBase]:
        return cls._TASK_CLASSES
```

Reply:
I've implemented this property and also `ModuleBase.get_inference_signatures` to replace this use of `_TASK_INFERENCE_SIGNATURES`.